en/scripting/modules/example.md

```diff
@@ -0,0 +1,248 @@
+
+- Look at the `main` field in the `package.json` file and determine the entry file `index.js`.
+- Look at the `type` field in the `package.json` file and notice that there is no `type` field.
+
+Identify the module format, it can be inferred that this is a **CJS** module. By the way, you can see that each js file in the package corresponds to a .d.ts file, which means that the `protobufjs` package comes with a `TypeScript` declaration file, which makes it easy to import the `protobufjs` module and get the internal interface through code hints.
+
+Next, you can see it exported in ``index.js``.
+
+```js
+"use strict";
+module.exports = require("./src/index");
+```
+
+Determine the module format and export method. The next step is how to use the `protobufjs` module in the script assets.
+
+First, create a `test.ts` script under `assets`. Then, write the following code in the header of the script.
+
+```ts
+// Most npm modules can be used by directly importing the module name.
+import protobufjs from 'protobufjs';
+console.log(protobufjs);
+```
+
+When run in Chrome, the console output is as follows:
+
+![protobufjs-print-default](./example/protobufjs-print-default.png)
+
+Some developers may have received an error when writing `import protobufjs from 'protobufjs'`, indicating that the module `has no default export`. This is because **CJS** does not have `default` export, and **ESM** and **CJS** interact with each other when is treated as `module.exports` **as** `export default` and does not mean that **CJS** modules have true default exports. Therefore, to keep the original way of writing, it is possible to configure TypeScript, see [TypeScript configuration](#TypeScript-configuration).
+
+Next, you can use all the modules provided by `protobufjs` directly. Of course, if you only need specific submodule functionality, such as `light` and `minimal`, you can import the subpaths directly from the package.
+
+```ts
+// Using the light version
+import protobuf from 'protobufjs/light.js';
+// Using the minimal version
+import protobuf from 'protobufjs/minimal.js';
+```
+
+> **Note**: in the case of protobufjs and many classic npm packages, the suffix is required when importing subpaths in the package. Please refer to the [Cocos Creator module specification](./spec.md).
+
+#### TypeScript configuration
+
+When importing modules without default export in TypeScript, it usually appears that `Module '"/${project_path}/protobufjs"' has no default export.`. This is because currently `protobufjs` only provides CommonJS modules, and Cocos Creator accesses CommonJS modules via "default import", but CommonJS modules really don't have a "default export". In this case, you can set the `"allowSyntheticDefaultImports"` option in the `"compilerOptions"` field to `true` by editing the `tsconfig.json` file in the project directory. If the field is not present, you can fill it in yourself.
```

Comment:
```suggestion
When importing modules without default export in TypeScript, it usually appears that `Module '"/${project_path}/protobufjs"' has no default export.`. This is because currently `protobufjs` only provides CommonJS modules, and Cocos Creator accesses CommonJS modules via "default import", but CommonJS modules really don't have a "default export". In this case, set the `"allowSyntheticDefaultImports"` option in the `"compilerOptions"` field to `true` by editing the `tsconfig.json` file in the project directory. If the field is not present, manually fill it in.
```